src/main/scala/com/timeout/kamon/cloudwatch/FutureUtils.scala

```diff
@@ -0,0 +1,39 @@
+package com.timeout.kamon.cloudwatch
+
+
+import java.util.concurrent.{CancellationException, Future => JavaFuture}
+
+import scala.annotation.tailrec
+import scala.concurrent.{ExecutionContext, Future, Promise}
+import scala.util.Try
+
+
+object FutureUtils {
+
+  implicit class FutureOps[T](jFuture: JavaFuture[T]) {
```

Comment:
Its probably easier / better to handle Amazon async commands by passing in an  `AsyncHandler` that completes a promise